### PR TITLE
W 12006339 v25 dwfunction chk duke

### PR DESCRIPTION
**Pull request by dukesphere**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/dw-binaries-functions-concatwith.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+== concatWith&#40;source: Binary, with: Null&#41;: Binary
+
+Helper Function to make `concatWith` null Friendly
+
+
+[[concatwith3]]
+== concatWith&#40;source: Null, with: Binary&#41;: Binary
+
+Helper Function to make `concatWith` null Friendly
+
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-binaries-functions-concatwith.adoc:56]: done: copyedited all in source

File: modules/ROOT/pages/dw-binaries-functions-tobase64.adoc [hunk 2]
```diff
@@ -18,8 +18,6 @@ Transforms a binary value into a Base64 string.
 | `content` | The binary value to transform.
 |===
 
-[[toBase64-example]]
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-binaries-functions-tobase64.adoc:21]: not needed anymore

**Review by dukesphere — COMMENTED** [review]
initial comments

File: modules/ROOT/pages/dw-coercions-functions-toarray.adoc [hunk 3]
```diff
@@ -18,7 +18,7 @@ _Introduced in DataWeave version 2.4.0._
 |===
 | Name | Description
 | `text` | The `String` value to transform into an `Array`
-           of characters (a `Array<String>` type).
+         of characters (a `Array<String>` type).
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-coercions-functions-toarray.adoc:21]: fixed in https://github.com/mulesoft/data-weave/pull/1977/files

File: modules/ROOT/pages/dw-binaries-functions-concatwith.adoc (hunk 1 above)
Comment (dukesphere) [reply modules/ROOT/pages/dw-binaries-functions-concatwith.adoc:56]: fixed in https://github.com/mulesoft/data-weave/pull/1977/files

File: modules/ROOT/pages/dw-core-annotations.adoc [hunk 4]
```diff
@@ -57,8 +54,6 @@ _Introduced in DataWeave version 2.4.0._
 marked annotation will wrap an annotated function with an `interceptorFunction`.
 An example is the `RuntimePrivilege` annotation, which is annotated by
 `@Interceptor(interceptorFunction = "@native system::SecurityManagerCheckFunctionValue")`.
-The `readUrl` function definition is annotated by `@RuntimePrivilege(requires = "Resource")`.
```
[the remark this answers is not in the document]
Comment (afelisatti) [reply modules/ROOT/pages/dw-core-annotations.adoc:60]: Yes, it marks the security privileges required for the function to work.

[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-annotations.adoc:60]: Not sure why it is not in the current source. Will restore there.

File: modules/ROOT/pages/dw-coercions-functions-tostring.adoc [hunk 5]
```diff
@@ -21,20 +21,25 @@ _Introduced in DataWeave version 2.4.0._
...
 | `roundMode` | Optional parameter for rounding decimal values
-                when the formatting presents a rounding choice,
-                such as a format of `0.#` for the decimal `0.15`.
-                The default is `HALF_UP`, and a `null` value
-                returns behaves like `HALF_UP`.
-                Only one of the following values is permitted:
+              when the formatting presents a rounding choice,
+              such as a format of `0.#` for the decimal `0.15`.
+              The default is `HALF_UP`, and a `null` value
+              returns behaves like `HALF_UP`.
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-coercions-functions-tostring.adoc:41]: @afelisatti Actually, I don't understand the comment about a null value here. If you attempt to coerce null to a string, you get an error.

File: modules/ROOT/pages/dw-core-annotations.adoc [hunk 6]
```diff
@@ -6,25 +6,24 @@
...
+apply to any valid target.
+
 
 Annotation Targets:
 
-* `Parameter`: For function parameters.
-* `Function`: For function definitions.
-* `Variable`: For variable definitions.
-* `Import`: For import definitions.
+* Parameter: For function parameters.
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-annotations.adoc:18]: done

File: modules/ROOT/pages/dw-core-annotations.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-annotations.adoc:60]: restored

File: modules/ROOT/pages/dw-core-functions-filterobject.adoc [hunk 7]
```diff
@@ -95,38 +95,9 @@ output application/json
 { "1": "a" }
 ----
 
-=== Example
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-functions-filterobject.adoc:98]: restored

File: modules/ROOT/pages/dw-core-functions-groupby.adoc [hunk 8]
```diff
@@ -6,7 +6,7 @@ endif::[]
 
 
 [[groupby1]]
-== groupBy<T, R&#62;&#40;items: Array<T&#62;, criteria: &#40;item: T, index: Number&#41; &#45;&#62; R&#41;: { &#40;R&#41;: Array<T&#62; }
+== groupBy<T, R&#62;&#40;items: Array<T&#62;, criteria: &#40;item: T, index: Number&#41; &#45;&#62; R&#41;: {&#124; &#40;R&#41;: Array<T&#62; &#124;}
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-functions-groupby.adoc:9]: meaning `&#124;` resolve to pipes. What do these mean in the signature?

File: modules/ROOT/pages/dw-core-functions-minusminus.adoc [hunk 9]
```diff
@@ -15,11 +15,11 @@ This version of `--` removes all instances of the specified items from an array.
 versions act on objects, strings, and the various date and time formats that
 are supported by DataWeave.
 
-[%header, cols="1,3"]
+[source,JSON,linenums]
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-functions-minusminus.adoc:18]: fixed

File: modules/ROOT/pages/dw-core-functions-reduce.adoc [hunk 10]
```diff
@@ -25,7 +25,7 @@ accumulator parameter, a null value is returned.
 |===
 | Name | Description
 | `item` | Item in the input array. It provides the value to reduce. Can also be referenced as `&#36;`.
-| `acc` | The accumulator. Can also be referenced as `&#36;&#36;`. Used to store the result of the lambda expression after each iteration of the `reduce` operation.
+| `acc` | The accumulator. Can also be referenced as `&#36;&#36;`. Used to store the result of the lambda expression after each iteration of the reduce operation.
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-functions-reduce.adoc:28]: done

File: modules/ROOT/pages/dw-core-functions-typeof.adoc [hunk 11]
```diff
@@ -8,14 +8,20 @@ endif::[]
...
+alias. If the value has metadata present, this will be included on the result of
+`typeOf`.
 
 === Parameters
 
 [%header, cols="1,3"]
 |===
 | Name | Description
-| `value` | A string, object, array, number, or other supported type.
+| `value` | A value.
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-core-functions-typeof.adoc:24]: done

File: modules/ROOT/pages/dw-core-functions-typeof.adoc [hunk 12]
```diff
@@ -8,14 +8,20 @@ endif::[]
 [[typeof1]]
 == typeOf<T&#62;&#40;value: T&#41;: Type<T&#62;
 
-Returns the type of a value.
+Returns the basic type of a value.
+
+
+A value's type is taken from its runtime representation, and will never be one of
+the arithmetic types (intersection, union, `Any`, or `Nothing`) nor a type
+alias. If the value has metadata present, this will be included on the result of
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-core-functions-typeof.adoc:16]: clarified "If the value has metadata present, this..."

File: modules/ROOT/pages/dw-dates-functions-atbeginningofyear.adoc [hunk 13]
```diff
@@ -26,7 +26,7 @@ _Introduced in DataWeave version 2.4.0._
 
 === Example
 
-This example transforms the `DateTime` input (`|2020-10-06T18:23:20.351-03:00|`)
+This example transforms the `DateTime` input (`&#124;2020-10-06T18:23:20.351-03:00&#124;`)
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-dates-functions-atbeginningofyear.adoc:29]: looks good in the output, pipes appear around datetime, as expected

File: modules/ROOT/pages/dw-dates-functions-datetime.adoc [hunk 14]
```diff
@@ -8,7 +8,7 @@ endif::[]
 [[datetime1]]
 == dateTime&#40;parts: DateTimeFactory&#41;: DateTime
 
-Creates a `DateTime` value from values specified for `year`, `month`, `day`, `hour`,
+Creates a `DateTime` value from values specified for `month`, `day`, `hour`,
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-dates-functions-datetime.adoc:11]: added back based on example

File: modules/ROOT/pages/dw-dates.adoc [hunk 15]
```diff
@@ -11,7 +11,7 @@ _Introduced in DataWeave version 2.4.0._
 
 == Functions
 
-[%header, cols="1,3a"]
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-dates.adoc:14]: @afelisatti can we update the parser to make cols value look like above, specifically with the 3a for multiline content in the table?

[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-dates.adoc:14]: Basically on all pages that have the == Functions heading for the list of functions in the module

File: modules/ROOT/pages/dw-runtime-functions-version.adoc [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+
+
+[[version1]]
+== version&#40;&#41;: String
+
+Returns the DataWeave version that its currently running.
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-runtime-functions-version.adoc:11]: copyedited entire version function content

File: modules/ROOT/pages/dw-runtime-types.adoc [hunk 17]
```diff
@@ -23,8 +15,8 @@ _Experimental:_ This type is an experimental feature that is subject to change o
 | `type DataFormatProperty = { name: String, optional: Boolean, defaultValue?: Any, description: String, possibleValues: Array<Any&#62; }`
 | Type that describes a data format property. The fields include a `name`,
 `description`, array of possible values (`possibleValues`), an optional default
- value (`defaultValue`), and an `optional` flag that indicates whether the
- property is required or not.
+value (`defaultValue`), and  an `optional` flag that indicates whether the property
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-runtime-types.adoc:18]: fixed extra space

File: modules/ROOT/pages/dw-strings-functions-countmatches.adoc [hunk 18]
```diff
@@ -30,7 +30,7 @@ This example counts matches in a string.
 [source,DataWeave,linenums]
 ----
 %dw 2.0
-import * from dw::core::Strings
+import countMatches from dw::core::Strings
```
[the remark this answers is not in the document]
Comment (dukesphere) [reply modules/ROOT/pages/dw-strings-functions-countmatches.adoc:33]: later, no urgent

**Comment by dukesphere** [comment]
replaced and updated by https://github.com/mulesoft/docs-dataweave/pull/226/files

**Pull request closed without merging**